### DISCUSSION TITLE
Control points registration

### DISCUSSION THREAD

**Discussion by domod72** [openMVG/openMVG#1974] [Q&A] [answered]
Hi
1. Is there a way to run control points registration from command line passing in input a file with points coordinates?
2. Does "control points registration tool" perform a compensation in case of non linear distortion of my model?
Thank you for your answer.

**Comment by pmoulon** [comment, marked as answer]
We don't have yet a command-line tool for this, but you could help us to move the logic from here MainWindow::registerProject() https://github.com/openMVG/openMVG/blob/develop/src/software/ui/SfM/control_points_registration/MainWindow.cpp#L207 to a function, so we could use for the UI and a command-line tool.

Regarding point 2. We perform first a rigid registration and then compute a Bundle Adjustment with GCP https://github.com/openMVG/openMVG/blob/develop/src/software/ui/SfM/control_points_registration/MainWindow.cpp#L377 but setting the right weight is tricky (by default we use a value of 20 to set some emphasis on the residual linked to the GCP).

**Reply by domod72** [reply]
Thank you for your answer.
I will take a look as soon as possible.
